src/browser/stacktrace.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    ShallowStackFrames,
+    applyStackFrames,
+    captureRawStackFrames,
+    filterExtraWdioFrames,
+} from "../utils/stacktrace";
+import { getBrowserCommands, getElementCommands } from "./history/commands";
+import { runWithHooks } from "./history/utils";
+
+type AnyFunc = (...args: any[]) => unknown; // eslint-disable-line @typescript-eslint/no-explicit-any
+
+export const runWithStacktraceHooks = ({
+    stackFrames,
+    fn,
+    stackFilterFunc,
+}: {
+    stackFrames: ShallowStackFrames;
+    fn: AnyFunc;
+    stackFilterFunc?: AnyFunc;
+}): ReturnType<typeof fn> => {
+    const frames = captureRawStackFrames(stackFilterFunc || runWithStacktraceHooks);
+
+    if (stackFrames.isNested(frames)) {
+        return fn();
+    }
```

Comment:
We dont want to save all of the frames, because with nested calls it would use some memory